imap_processing/swe/l1a/swe_science.py

```diff
@@ -139,99 +137,70 @@ def swe_science(decom_data):
+        name="polar_angle",
+        dims=["polar_angle"],
+        attrs=cdf_attrs.get_variable_attributes("polar_angle"),
     )
 
-    science_attrs = dataclasses.replace(
-        swe_cdf_attrs.l1a_science_attrs,
-        catdesc="Uncompressed counts from SWE",
-        fieldname="Uncompressed counts from SWE",
-        label_axis="Uncompressed counts from SWE",
-        units="counts",
-    )
     science_xarray = xr.DataArray(
         science_array,
-        dims=["epoch", "Energy", "Counts"],
-        attrs=science_attrs.output(),
+        dims=["epoch", "spin_angle", "polar_angle"],
+        attrs=cdf_attrs.get_variable_attributes("science_data"),
     )
 
-    raw_science_attrs = dataclasses.replace(
-        swe_cdf_attrs.l1a_science_attrs,
-        catdesc="Raw counts from SWE",
-        fieldname="Raw counts from SWE",
-        label_axis="Raw counts from SWE",
-        units="counts",
-    )
     raw_science_xarray = xr.DataArray(
         raw_science_array,
-        dims=["epoch", "Energy", "Counts"],
-        attrs=raw_science_attrs.output(),
+        dims=["epoch", "spin_angle", "polar_angle"],
+        attrs=cdf_attrs.get_variable_attributes("raw_counts"),
     )
 
+    # Add APID to global attrs for following processing steps
+    l1a_global_attrs = cdf_attrs.get_global_attributes("imap_swe_l1a_sci")
+    # Formatting to string to be complaint with ISTP
+    l1a_global_attrs["packet_apid"] = f"{decom_data[0].header['PKT_APID'].raw_value}"
     dataset = xr.Dataset(
         coords={
             "epoch": epoch_time,
-            "Energy": energy,
-            "Counts": counts,
+            "spin_angle": spin_angle,
+            "polar_angle": polar_angle,
         },
-        attrs=swe_cdf_attrs.swe_l1a_global_attrs.output(),
+        attrs=l1a_global_attrs,
     )
-    dataset["SCIENCE_DATA"] = science_xarray
-    dataset["RAW_SCIENCE_DATA"] = raw_science_xarray
+    dataset["science_data"] = science_xarray
+    dataset["raw_science_data"] = raw_science_xarray
 
     # create xarray dataset for each metadata field
     for key, value in metadata_arrays.items():
-        # TODO: figure out how to add more descriptive
-        # description for each metadata field
-        #
-        # int_attrs["CATDESC"] = int_attrs["FIELDNAM"] = int_attrs["LABLAXIS"] = key
-        # # get int32's max since most of metadata is under 32-bits
-        # int_attrs["VALIDMAX"] = np.iinfo(np.int32).max
-        # int_attrs["DEPEND_0"] = "epoch"
-        dataset[key] = xr.DataArray(
+        # Lowercase the key to be complaint with ISTP's metadata field
+        metadata_field = key.lower()
+        dataset[metadata_field] = xr.DataArray(
             value,
             dims=["epoch"],
-            attrs=dataclasses.replace(
-                swe_cdf_attrs.swe_metadata_attrs,
-                catdesc=key,
-                fieldname=key,
-                label_axis=key,
-                depend_0="epoch",
-            ).output(),
```

Comment:
I love seeing how much code is removed here 😆 so satisfying!

Reply:
same here!